components/engine_rocks/src/rocks_metrics.rs

```diff
@@ -910,206 +911,274 @@ pub fn flush_engine_histogram_metrics(t: HistType, value: HistogramData, name: &
+            let handle = crate::util::get_cf_handle(db, cf).unwrap();
+            // It is important to monitor each cf's size, especially the "raft" and "lock"
+            // column families.
+            cf_stats.used_size += crate::util::get_engine_cf_used_size(db, handle);
+            cf_stats.blob_cache_size += db.get_blob_cache_usage_cf(handle);
 
-        // Pending compaction bytes
-        if let Some(pending_compaction_bytes) =
-            crate::util::get_cf_pending_compaction_bytes(engine, handle)
-        {
-            STORE_ENGINE_PENDING_COMPACTION_BYTES_VEC
-                .with_label_values(&[name, cf])
-                .set(pending_compaction_bytes as i64);
-        }
+            // TODO: find a better place to record these metrics.
+            // Refer: https://github.com/facebook/rocksdb/wiki/Memory-usage-in-RocksDB
+            // For index and filter blocks memory
+            cf_stats.readers_mem += db
+                .get_property_int_cf(handle, ROCKSDB_TABLE_READERS_MEM)
+                .unwrap_or(0);
+            cf_stats.mem_tables += db
+                .get_property_int_cf(handle, ROCKSDB_CUR_SIZE_ALL_MEM_TABLES)
+                .unwrap_or(0);
+            // TODO: add cache usage and pinned usage.
+            cf_stats.num_keys += db
+                .get_property_int_cf(handle, ROCKSDB_ESTIMATE_NUM_KEYS)
+                .unwrap_or(0);
+            cf_stats.pending_compaction_bytes +=
+                crate::util::get_cf_pending_compaction_bytes(db, handle).unwrap_or(0);
+            cf_stats.num_immutable_mem_table +=
+                crate::util::get_cf_num_immutable_mem_table(db, handle).unwrap_or(0);
+            // Titan.
+            cf_stats.live_blob_size += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_LIVE_BLOB_SIZE)
+                .unwrap_or(0);
+            cf_stats.num_live_blob_file += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_NUM_LIVE_BLOB_FILE)
+                .unwrap_or(0);
+            cf_stats.num_obsolete_blob_file += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_NUM_OBSOLETE_BLOB_FILE)
+                .unwrap_or(0);
+            cf_stats.live_blob_file_size += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_LIVE_BLOB_FILE_SIZE)
+                .unwrap_or(0);
+            cf_stats.obsolete_blob_file_size += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_OBSOLETE_BLOB_FILE_SIZE)
+                .unwrap_or(0);
+            cf_stats.blob_file_discardable_ratio_le0 += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_DISCARDABLE_RATIO_LE0_FILE)
+                .unwrap_or(0);
+            cf_stats.blob_file_discardable_ratio_le20 += db
+                .get_property_int_cf(handle, ROCKSDB_TITANDB_DISCARDABLE_RATIO_LE20_FILE)
+                .unwrap_or(0);
+            cf_stats.blob_file_discardable_ratio_le50 += db
```

Comment:
I'm worried about whether 0 is better than `None`, as we have done a lot of work to reduce the size of statistics.